state/carmen.go

```diff
@@ -160,6 +164,21 @@ func (s *carmenStateDB) SetState(addr common.Address, key common.Hash, value com
 	s.txCtx.SetState(carmen.Address(addr), carmen.Key(key), carmen.Value(value))
 }
 
+func (s *carmenStateDB) GetStorageRoot(addr common.Address) common.Hash {
+	panic("GetStorageRoot not implemented")
+	return common.Hash{}
+}
```

Comment:
fyi: We agreed that we actually dont need this func. We only need an information whether the storage is empty or not. The func will be `HasEmptyStorage(addr) bool`. A PR with this change is currently under review.

Reply:
GetStorageRoot returns a hash. What should be returned if `HasEmptyStorage(addr) bool` is used instead?

Reply:
I think the correct way to return an empty hash for `true` and non-empty for `false`. Afaik @HerbertJordan said that the way this func is utilized is only check for emptiness.

Reply:
That's right. `GetStorageRoot` is needed [here](https://github.com/Fantom-foundation/go-ethereum-sonic/blob/2400937cc3b1c370ea8735d7af86eac621e4eb70/core/vm/evm.go#L480) only to be compared with either the zero-hash or the hash of an empty trie.

So, to implement this function it should be sufficient to call `HasEmptySotrage(addr)` and return `Hash{0}` if the result is `true` and `Hash{1}` if the result is `false`.

Reply:
Please see #1119